pkg/providers/aws/provider_postgres.go

```diff
@@ -700,11 +700,12 @@ func buildRDSUpdateStrategy(rdsConfig *rds.CreateDBInstanceInput, foundConfig *r
 			return nil, errorUtil.Wrap(err, "invalid postgres version")
 		}
 		if engineUpgradeNeeded {
+			logrus.Info(fmt.Sprintf("Engine upgrade found, the current EngineVersion is %s and is upgrading to %s", *foundConfig.EngineVersion, *rdsConfig.EngineVersion))
 			mi.EngineVersion = rdsConfig.EngineVersion
 			if cr.Spec.ApplyImmediately {
 				mi.ApplyImmediately = aws.Bool(cr.Spec.ApplyImmediately)
 			}
-			updateFound = true
+			return mi, nil
```

Comment:
should this be inside the if statement? in this case we return that an update is required even when it may not be.

Reply:
or alternatively we add a condition to line 711 where we don't return nil if applyimmediately is true

Reply:
ah okay I see the `engineUpgradeNeeded` if statement now. ignore the above. 👍

Reply:
@austincunningham after some further digging this will have the api called on every reconcile. I think it would be better to put his change into the check on engineupgrade in the verifyPendingModifications function. what we want to do is allow for an additional api call if engineupgrade is sitting in pendingmodifications on line 739 we check if the modifyInput object is equal tot he existing pending modifications. if that's true we don't call another update. so we could pass down the applyImmediately value and if it's true and there is a pending modification then return true rather than false

Reply:
as well as that we should revert this line to prevent sending a request to the api on every reconcile until the version is upgraded.

Reply:
This suggestion is outdated by the more recent one so resolving.